### PR TITLE
patch kartograf to set min version of `gufe`

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist

* [x] Used a static YAML file for the patch if possible ([instructions](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/README.md)).
* [x] Only wrote code directly into `generate_patch_json.py` if absolutely necessary.
* [x] Ran `pre-commit run -a` and ensured all files pass the linting checks.
* [x] Ran `python show_diff.py` and posted the output as part of the PR.
* [x] Modifications won't affect packages built in the future. 

See https://github.com/conda-forge/kartograf-feedstock/issues/10
tl;dr we didn't have a min pin so really old packages of gufe could get pulled down

### CODE REVIEW

**Comment by mikemhenry** [comment]
```
$ python show_diff.py
================================================================================
================================================================================
linux-armv7l
================================================================================
================================================================================
win-32
================================================================================
================================================================================
osx-arm64
================================================================================
================================================================================
linux-ppc64le
================================================================================
================================================================================
linux-aarch64
================================================================================
================================================================================
noarch
noarch::kartograf-1.0.0-pyhd8ed1ab_1.conda
noarch::kartograf-1.0.0-pyhd8ed1ab_2.conda
noarch::kartograf-0.5.2-pyhd8ed1ab_1.conda
noarch::kartograf-0.0.6-pyhd8ed1ab_0.conda
noarch::kartograf-0.5.2-pyhd8ed1ab_0.conda
-    "gufe",
+    "gufe>=0.9.5",
================================================================================
================================================================================
win-64
================================================================================
================================================================================
osx-64
================================================================================
================================================================================
linux-64
```

**Review by xhochy — APPROVED** [review]
